lsvmutils/blkdev.c

```diff
@@ -51,21 +51,36 @@ int BlkdevRead(
     ptr = (UINT8*)data;
     rem = size;
 
-    for (i = blkno; rem > 0 && i < blkno + nblocks; i++)
+    /* Nothing to do. */
+    if (nblocks == 0)
     {
-        UINT8 block[BLKDEV_BLKSIZE];
-        UINTN n;
+        rc = 0;
+        goto done;
+    }
 
-        if (dev->Get(dev, i, block) != 0)
+    /* If size is aligned to block size, we just do a batch read. */
```

Comment:
I would probably say "If the size is a multiple of the block size".

Reply:
Fixed